src/pages/index.js

```diff
@@ -13,72 +15,100 @@ import {
-//   },
-//   {
-//     name: "Bald Mountains",
-//     link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/bald-mountains.jpg",
-//   },
-//   {
-//     name: "Latemar",
-//     link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/latemar.jpg",
-//   },
-//   {
-//     name: "Vanoise National Park",
-//     link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/vanoise.jpg",
-//   },
-//   {
-//     name: "Lago di Braies",
-//     link: "https://practicum-content.s3.us-west-1.amazonaws.com/software-engineer/around-project/lago.jpg",
-//   },
-// ];
-
-// ---------------- DOM Element References ----------------
-// const profileEditForm = document.forms["profile-form"];
-// const addCardForm = document.forms["card-form"];
-// const profileEditModal = document.querySelector("#profile-edit-modal");
-// const addCardModal = document.querySelector("#add-card-modal");
-// const previewImageModal = document.querySelector("#preview-image-modal");
-
-// const cardTitleInput = document.querySelector("#card-title");
-// const cardUrlInput = document.querySelector("#card-url");
-// const cardsListEl = document.querySelector(".cards__list");
-// const previewImage = previewImageModal.querySelector(".modal__image");
-// const previewCaption = previewImageModal.querySelector(".modal__caption");
-// const modalOpenedSelector = "modal_opened";
+import Popup from "../components/Popup.js";
 
 // ---------------- User Info ----------------
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "11f34e71-370c-42ab-866a-065e85c99efb",
+    "Content-Type": "application/json",
+  },
+});
+
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   infoSelector: ".profile__description",
-  // imageSelector: ".profile__image",
+  imageSelector: ".profile__image",
 });
 
+let section;
+
 // ---------------- Card Functions ----------------
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteIcon,
+    handleLikeButtonClick
+  );
+  console.log(cardData);
   return card.generateCard();
 }
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: addCard,
-  },
-  ".cards__list"
-);
-section.renderItems();
+const confirmDeletePopup = new confirmDelete("#delete-popup", handleFormDelete);
+confirmDeletePopup.setEventListeners();
+
+function handleFormDelete(card) {
+  console.log("Form was submitted");
+  deleteCard(card._id)
+    .then((response) => {
+      console.log("Card deleted successfully", response);
+      closePopup();
+    })
+    .catch((error) => {
+      console.error("Error deleting card", error);
+    });
```

Comment:
This code doesn't work. It should be just an empty function because it's reasigned by the `confirmDelete` method on opening


Like this:

```js
function handleFormDelete() {}


```